.github/workflows/performance.yml

```diff
@@ -0,0 +1,151 @@
+# This is a workflow triggered by PR or triggered manually
+# Runs quick performance tests and reports the comparison against HEAD
+# Test should take less than 10 minutes to run on current self-hosted devices
+name: "Performance Testing"
+
+# Controls when the action will run.
+# Workflow runs when manually triggered using the UI or API.
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+# Env variables
+env:
+  SG_COMMIT: 649509c
+
+jobs:
+  Performance_x86-64:
+    name: Performance x86-64
+    runs-on: [self-hosted, linux, x64]
+    # Inputs the workflow accepts.
+    steps:
+      - run: echo "This job is now running on a ${{ runner.os }} self-hosted server."
+
+      - name: Setup Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```

Comment:
Personally I'd prefer to keep the usage of external actions to a minimum for the wasmtime repo, so here if we can arrange for rustup to be installed on the hosted runner this I think could be `rustup update stable && rustup default stable`